### PR TITLE
 Kafka 2.4 + KIP-455 for Xinfra Monitor Multie Cluster Topic Management Service 

### PR DESCRIPTION
**Pull request by andrewchoi5**
`Kafka 2.4` + KIP-455 for Xinfra Monitor `Multi Cluster Topic Management Service` 

Total extends CumulativeSum : A non-sampled cumulative total maintained over all time.
 This is a non-sampled version of {@link WindowedSum}.
Total is deprecated since 2.4. Recommended to use {@link CumulativeSum} instead.

(New) CumulativeSum: A non-sampled cumulative total maintained over all time.
This is a non-sampled version of {@link WindowedSum}.

### CODE REVIEW

File: src/main/java/com/linkedin/xinfra/monitor/services/MultiClusterTopicManagementService.java [hunk 1]
```diff
@@ -426,13 +431,12 @@ void maybeReassignPartitionAndElectLeader() throws Exception {
               "Configured replication factor {} is smaller than the current replication factor {} of the topic {} in cluster.",
               _replicationFactor, currentReplicationFactor, _topic);
 
-        if (expectedReplicationFactor > currentReplicationFactor && !zkClient
-            .reassignPartitionsInProgress()) {
+        if (expectedReplicationFactor > currentReplicationFactor
+            && Utils.ongoingPartitionReassignments(_adminClient).isEmpty()) {
```
Comment (gitlw) [inline src/main/java/com/linkedin/xinfra/monitor/services/MultiClusterTopicManagementService.java:435]: Can we check the ongoing partition reassignments only for this single topic?

Comment (gitlw) [reply src/main/java/com/linkedin/xinfra/monitor/services/MultiClusterTopicManagementService.java:435]: I mean passing the list of partitions, and call the API 
listPartitionReassignments(Set<TopicPartition> partitions)

Comment (andrewchoi5) [reply src/main/java/com/linkedin/xinfra/monitor/services/MultiClusterTopicManagementService.java:435]: We can. We wanted to replicate the previous behavior of `.reassignPartitionsInProgress()` which checks for all. 
Do we want to check only a set of topic partitions? @gitlw

Comment (andrewchoi5) [reply src/main/java/com/linkedin/xinfra/monitor/services/MultiClusterTopicManagementService.java:435]: I think you raise a very good point but in order to keep the exact same behaviour to be on a much safer side, we should probably keep the Utils.ongoingPartitionReassignments.isEmpty? — which is similar to  !zkClient.reassignPartitionsInProgress()
Let's check if getting all of the ongoing partition reassignments is costly when using the new API.

**Review by gitlw — APPROVED** [review]
LGTM

**Pull request merged**